pkg/certificates/reconciler/controller.go

```diff
@@ -45,13 +46,13 @@ func NewControllerFactory(componentName string) injection.ControllerConstructor
 		ctx context.Context,
 		cmw configmap.Watcher,
 	) *controller.Impl {
-		ctx = logging.WithLogger(ctx, logging.FromContext(ctx).Named("ctrl-secrets-controller"))
-
-		secretInformer := secretinformer.Get(ctx)
 
 		caSecretName := componentName + caSecretNamePostfix
 		labelName := componentName + secretLabelNamePostfix
 
+		ctx = filteredFactory.WithSelectors(ctx, labelName)
+		secretInformer := getSecretInformer(ctx)
```

Comment:
In net-kourier, istio this is behind an [flag/env variable](https://github.com/knative-sandbox/net-kourier/blob/main/pkg/reconciler/informerfiltering/util.go). 
The plan was to enable by default at some point but the idea was to do at some release so we don't break stuff. Should we align all code bases for 1.9? cc @dprotaso

Reply:
my assumption here is that it should not be watching on all secrets without a label because this causes a load on memory , so added it in all cases.

I can add the flag if needed